web_programming/convert_number_to_words.py

```diff
@@ -1,106 +1,83 @@
 import math
 
 
-def convert(number: int) -> str:
-    """
-    Given a number return the number in words.
+def int_to_en(num):
+    d = {
+        0: "zero",
+        1: "one",
+        2: "two",
+        3: "three",
+        4: "four",
+        5: "five",
+        6: "six",
+        7: "seven",
+        8: "eight",
+        9: "nine",
+        10: "ten",
+        11: "eleven",
+        12: "twelve",
+        13: "thirteen",
+        14: "fourteen",
+        15: "fifteen",
+        16: "sixteen",
+        17: "seventeen",
+        18: "eighteen",
+        19: "nineteen",
+        20: "twenty",
+        30: "thirty",
+        40: "forty",
+        50: "fifty",
+        60: "sixty",
+        70: "seventy",
+        80: "eighty",
+        90: "ninety",
+    }
+    k = 1000
+    m = k * 1000
+    b = m * 1000
+    t = b * 1000
 
-    >>> convert(123)
-    'OneHundred,TwentyThree'
-    """
-    if number == 0:
-        words = "Zero"
-        return words
-    else:
-        digits = math.log10(number)
-        digits = digits + 1
-        singles = {}
-        singles[0] = ""
-        singles[1] = "One"
-        singles[2] = "Two"
-        singles[3] = "Three"
-        singles[4] = "Four"
-        singles[5] = "Five"
-        singles[6] = "Six"
-        singles[7] = "Seven"
-        singles[8] = "Eight"
-        singles[9] = "Nine"
+    assert 0 <= num
+
+    if num < 20:
+        return d[num]
 
-        doubles = {}
-        doubles[0] = ""
-        doubles[2] = "Twenty"
-        doubles[3] = "Thirty"
-        doubles[4] = "Forty"
-        doubles[5] = "Fifty"
-        doubles[6] = "Sixty"
-        doubles[7] = "Seventy"
-        doubles[8] = "Eighty"
-        doubles[9] = "Ninety"
+    if num < 100:
+        if num % 10 == 0:
+            return d[num]
+        else:
+            return d[num // 10 * 10] + "-" + d[num % 10]
```

Comment:
Let's use `divmod()`... https://docs.python.org/3/library/functions.html
```suggestion
        div, mod = divmod(num)
        return f"{d[div * 10]}-{d[mod]}" if mod else d[num]
```
Repeat below.